core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -46,7 +46,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   /** Create a SparkConf that loads defaults from system properties and the classpath */
   def this() = this(true)
 
-  private[spark] val settings = new HashMap[String, String]()
+  private[spark] val settings = new HashMap[String, String] with SynchronizedMap[String, String]
```

Comment:
Hmm. Any reason to use `SynchronizedMap` instead of a map whose implementation is thread-safe (such as `TrieMap` or one of the `java.util.concurrent` ones)?